src/main/scala/cromwell/parser/AstTools.scala

```diff
@@ -3,20 +3,43 @@ package cromwell.parser
 import java.io.File
 
 import cromwell.binding._
+import cromwell.binding.types._
 import cromwell.parser.WdlParser.{Ast, AstList, AstNode, Terminal}
 import cromwell.util.FileUtil
 
 import scala.collection.JavaConverters._
 
 object AstTools {
-  implicit class EnhancedAstNode(val ast: AstNode) extends AnyVal {
-    def findAsts(name: String): Seq[Ast] = AstTools.findAsts(ast, name)
+  implicit class EnhancedAstNode(val astNode: AstNode) extends AnyVal {
+    def findAsts(name: String): Seq[Ast] = AstTools.findAsts(astNode, name)
     def findAstsWithTrail(name: String, trail: Seq[AstNode] = Seq.empty): Map[Ast, Seq[AstNode]] = {
-      AstTools.findAstsWithTrail(ast, name, trail)
+      AstTools.findAstsWithTrail(astNode, name, trail)
+    }
+    def findTerminals(): Seq[Terminal] = AstTools.findTerminals(astNode)
+    def findTopLevelMemberAccesses(): Iterable[Ast] = AstTools.findTopLevelMemberAccesses(astNode)
+    def sourceString(): String = astNode.asInstanceOf[Terminal].getSourceString
+    
+    def wdlType: WdlType = {
+      astNode match {
+        case t: Terminal =>
+          t.getSourceString match {
```

Comment:
I had an issue in this function during rebase... I had added a syntax error in here for incorrect arrays, but then removed it since it was awkward to get the SyntaxErrorFormatter into this function.  Not too big of a deal right now but something to keep in mind I guess